app/application/use_cases/generate_provisioning_token.py

```
@@ -0,0 +1,46 @@
+@dataclass
+class GenerateProvisioningTokenInput:
+    user_id: str
+
+
+class GenerateProvisioningToken:
+    def __init__(self, user_repo: UserRepository) -> None:
+        self._user_repo = user_repo
+
+    def execute(self, input: GenerateProvisioningTokenInput) -> str:
+        for attempt in range(MAX_RETRIES):
+            digits = "".join(random.choices(string.digits, k=6))
```

Ruff (0.15.2)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)